### PR TITLE
amends #21690 to fix broken Nim to C++ source line mappings

### PR DESCRIPTION
**Pull request by heterodoxic**
This PR aims to restore most of the formatting changes I introduced with my #21690, which led to wrong Nim to C++ source line mappings. Many thanks for your report @planetis-m here: https://github.com/nim-lang/Nim/pull/21690#issuecomment-1528855789! My naive approach did not take into account that a better formatting together with `--lineDir:on` or `--debugger:native` will have to include more `#line` directives, sorry! I will try to achieve this in a later attempt.

As it stands, at least several redundant `#line` directives should be gone, and will only carry file information if the file has actually changed, also applied to Nim's stack frame info.

TL;DR: apologies for the trouble I caused ☹️, re-uglified most of the C++ debug output **for now**, WIP...

### CODE REVIEW

**Comment by Araq** [comment]
Please rebase.
